### DISCUSSION TITLE
Not a single which-key binding is registered

### DISCUSSION THREAD

**Discussion by lmburns** [mrjones2014/legendary.nvim#123] [Q&A] [answered]
I know that these bindings are being picked up by `legendary`, until they reach this [line](https://github.com/mrjones2014/legendary.nvim/blob/5b8c1611185265357cdba451aae4ad0730aee413/lua/legendary/compat/which-key.lua#L31). So I have two questions as a result of this.

1. Should that line instead check if the `group` field is nil instead of converting it to a string? Because a `nil` value is being shown as 3 characters in length (which is greater than 0), causing every binding to be skipped.
2. If this is to be the case and every single binding which has a group set (`true`, `false`, and `nil`) is to be skipped, how can I manually turn off the group for `which-key` to get them to register with `legendary`?

Note that the bindings I am talking about are the ones that are to be registered automatically, not in `legendary`'s `setup` call.

I've also tried setting a `group` field in each call to `which-key` I have in my configuration, but it doesn't seem to work. I've tried setting it to `nil`, `true`, `false`, a string, etc.

As a side note, I've also noticed that the `include_legendary_cmds` option does not turn off the `Legendary` commands from showing in the window. They are still there regardless of what the boolean value is.

Thanks

**Comment by mrjones2014** [comment, marked as answer]
Hey @lmburns sorry about that. I'm pretty sure that's a bug. I just pushed a new commit, can you see if that fixes it?

**Reply by lmburns** [reply]
Yes that does work, thank you.

Now, the `include_builtin` and the `include_legendary_cmds` don't seem to affect what they say they effect. Before this last change was committed, the `include_legendary_cmds` didn't seem to work either way; however, the `include_builtin` would not show the builtin prior to this.

**Reply by mrjones2014** [reply]
Sorry, can you clarify?

`include_legendary_cmds` didn't show them when it was `true`?

`include_builtin` is not including builtins for you, when set to `true`?

The change I just made is unrelated to either of those settings btw, but if you're having issues with them I'll definitely try to fix them.

**Reply by lmburns** [reply]
Yeah sorry for not being to clear.

Before this last commit:
- When `include_builtin` was false, it was in fact true that no builtin keymaps were shown. When it was true, they were shown.
- When `include_legendary_cmds` was false, the `:Legendary...` commands were shown, just as if the value were true. 

After this last commit:
- Whatever `include_builtin` is set to, the builtin keymaps are always shown.
- The same is with the `include_legendary_cmds` before and after the commit. They are shown regardless of what the boolean value is.

On another note, I've noticed that the `autocmds` I set with `legendary` are ran by `nvim`, but are not shown when listing them with `:Legendary autocmds`. They are neither shown when they are set on the `setup` function or if they are binded later on

**Comment by mrjones2014** [comment]
>When include_legendary_cmds was false, the :Legendary... commands were shown, just as if the value were true.

This is something I missed in the config setup. Thanks. I've pushed a commit to fix that.

As far as the rest of it, I'm unable to reproduce. Please file an issue with minimal configuration.

>On another note, I've noticed that the autocmds I set with legendary are ran by nvim, but are not shown when listing them with :Legendary autocmds. They are neither shown when they are set on the setup function or if they are binded later on

Did you set the `description` field on the autocmds? By default autocmds aren't shown unless they have a description because triggering autocmds manually is a much less common use case. [This is documented behavior](https://github.com/mrjones2014/legendary.nvim#table-structures).

**Reply by lmburns** [reply]
Ah, okay, yes, the setup I was using as a description was through the `opts` field, by setting a description which would be called on the actual API function `nvim_create_autocmd`. I had `opts = { desc = "description..." }`

Thank you, now it works.

**Comment by lmburns** [comment]
@mrjones2014 

To continue this in a similar fashion from yesterday, everything seems to be working fine and all which-key bindings are picked up, however, `legendary` believes that all bindings registered with `which-key` are in normal mode. This isn't the biggest deal in the world, but I'm just reporting another bug I've come across and am unsure if this is worth of creating a new issue.

Here is an example which will show the bindings as in normal mode (`n`) even though the keybinding was created for visual mode.

```lua
require("which-key").register(
    {
        ["d"] = {[["_d]], "Delete (blackhole)"},
    },
    {mode = "v"}
)
```

When creating a key-mapping with `legendary`, they are correctly shown. This for example shows
`n, i, x`.

```lua
require("legendary").bind_keymap(
    {
        "<C-,>",
        "<Cmd>Legendary keymaps<CR>",
        description = "Show Legendary keymaps",
        mode = {"n", "i", "x"}
    }
)
```

**Reply by mrjones2014** [reply]
Can you open an issue?
